FEATURE: Add support for AspNet Controller

ViniciusSouza opened the pull request:
This PR adds the support to call the ArribaManagementService using the ArribaController.

Using the AspNet framework will open space for new collaborators besides giving access to the features available.

ericmaino commented on src/Arriba/Arriba.Server/Program.cs:
```diff
@@ -64,158 +45,6 @@ private static IHostBuilder CreateHostBuilder(string[] args)
                 });
         }
-
-        private class Startup
```
This commit appears to mix a refactoring (code move/rename) with edits, which makes the edits much harder to detect and observe. I would recommend splitting this commit into at least two, but possibly more commits.

ericmaino commented on src/Arriba/Arriba.Server/Extensions/OAuthExtension.cs:
```diff
@@ -0,0 +1,35 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Arriba.Server.Extensions
```
Please update the default namespace for the project. The base namespace should be Arriba not Arriba.Server. This way this file falls into Arriba.Extensions instead of Arriba.Server.Extensions. You don't need to correct all other changes, but we should do that in another change soon.

ericmaino reviewed (commented):
The files moved into Configurations folder, should be moved into a folder called Configuration without the s and they should also have their namespaces updated.

ericmaino commented on src/Arriba/Arriba.Server/Controllers/ArribaController.cs:
```diff
@@ -0,0 +1,159 @@
+using System.Net;
+using System.Runtime.CompilerServices;
+
+namespace Arriba.Server.Controllers
```
Please correct project and namespace. This should be Arriba.Controllers

ericmaino commented on src/Arriba/Arriba.Server/Controllers/ArribaController.cs:
```diff
@@ -0,0 +1,159 @@
+            _arribaServerConfiguration = arribaServerConfiguration;
+        }
+
+        private IActionResult ExecuteAction(Action action, Func<IActionResult> result)
```
This feels unnecessary here, but instead should be moved into a middleware which should simplify this code a bit

ericmaino commented on src/Arriba/Arriba.Server/Controllers/ArribaController.cs:
```diff
@@ -0,0 +1,159 @@
+        {
+            if (ex is ArribaAccessForbiddenException)
+                if (_arribaServerConfiguration.EnabledAuthentication)
+                    return Forbid();
```
What's the difference between line 92 and 94?

ViniciusSouza replied:
When the Authentication is disabled if we use Forbid we end up getting an Exception because the didn't configure and use authorization. The forbid return 403 and redirects the user to the app login page, to avoid the exception I need to check and return Forbid when it is on and 403 when it is not.

ericmaino commented on src/Arriba/Arriba.Server/Controllers/ArribaController.cs:
```diff
@@ -0,0 +1,159 @@
+        private readonly IArribaManagementService _arribaManagement;
+        private readonly IArribaServerConfiguration _arribaServerConfiguration;
+
+        public ArribaController(IArribaManagementService arribaManagement, IArribaServerConfiguration arribaServerConfiguration)
```
Feels like IArribaServerConfiguration will go away if the exception handling is moved to a middleware

ericmaino commented on src/Arriba/Arriba.Server/Startup.cs:
```diff
@@ -46,8 +48,10 @@ public void ConfigureServices(IServiceCollection services)
+            var arribaManagementServiceFactory = new ArribaManagementServiceFactory(new DatabaseFactory().GetDatabase());
 
+            services.AddOAuth(serverConfig);
+            services.AddSingleton(arribaManagementServiceFactory.CreateArribaManagementService());
```
This does not feel necessary. Seems like the existing host construction could be used here. I agree we'll probably want to pick on and not have mixed DI implementations, but for now it feels like adding this extra factory is not necessary

ViniciusSouza committed:
PROJECT: Change default namespace to Arriba

ViniciusSouza committed:
FEATURE: Disabling authentication for developer loop

ViniciusSouza committed:
REFACTOR: Remove unused property

ViniciusSouza committed:
REFACTOR: Decouple Start from Program

ViniciusSouza committed:
REFACTOR: Removing unused code

ViniciusSouza committed:
REFACTOR: Start.Configure Use IHostEnvironment

ViniciusSouza committed:
REFACTOR: Move Configuration to Configuration folder

ViniciusSouza committed:
RENAME: ArribaContoller to ArribaController

ViniciusSouza committed:
FEATURE: Add Exception Middleware

ViniciusSouza committed:
FEATURE: Add ArribaResultFilter

ArribaRoutedApplication uses ArribaResponse that envelopes the service result.
This filter will envelop the result for ArribaController

ViniciusSouza committed:
FEATURE: Add new route at nginx for ArribaController

At this point the two implementation will run in parallel.
This way it is possible to check any odd behavior that may occur.

ViniciusSouza committed:
FEATURE: Implement ArribaController

ViniciusSouza committed:
REFACTOR: Use route datav2 at the backend url

Nginx has two routes data and datav2:
**data** uses the ApplicationRoutedTable services
**datav2** users the ArribaController services

The both routes will be running in parallel for some time.

ViniciusSouza committed:
REFACTOR: Using Arriba as namespace

Removing Server from the namespace.

ViniciusSouza committed:
FEATURE: Add OAuthExtension

ViniciusSouza committed:
REFACTOR: Use Extension OAuthExtension

ViniciusSouza committed:
REFACTOR: Split out host initalization

ericmaino committed:
FIX: Normalize api endpoints using versioning
